framework/Optimizers/Optimizer.py

```diff
@@ -12,250 +12,188 @@
-    #   "collecting grad eval points" - all the required gradient evaluation points are submitted, so we're just waiting to collect them
-    #   "submitting new opt points"   - a new optimal point has been postulated, and is being submitted for evaluationa (not actually used)
-    #   "collecting new opt points"   - the new  hypothetical optimal point has been submitted, and we're waiting on it to finish
-    #   "evaluate gradient"           - localStillReady notes we have all the new grad eval points, and has flagged for gradient to be evaluated in localGenerateInput
-    #   "following traj <#>"          - trajectory is following another one, given by the last word
-    # Reasons:
-    #   "just started"                - the optimizer has only just begun operation, and doesn't know what it's doing yet
-    #   "found new opt point"         - the last hypothetical optimal point has been accepted, so we need to move forward
-    #   "rejecting bad opt point"     - the last hypothetical optimal point was rejected, so we need to reconsider
-    #   "seeking new opt point"       - the process of looking for a new opt point has started
-    #   "converged"                   - the trajectory is in convergence
-    #   "removed as redundant"        - the trajectory has ended because it follows another one
-    #   "received recommended point"  - something other than the normal algorithm (such as a preconditioner) suggested a point
-    # example usage:
-    #   self.status[traj]['process'] == 'submitting grad eval points' and self.status[traj]['reason'] == 'rejecting bad opt point'
-    #
-    ### END explanation
-    self.addAssemblerObject('TargetEvaluation','1')
-    self.addAssemblerObject('Function','-1')
-    self.addAssemblerObject('Preconditioner','-n')
-    self.addAssemblerObject('Sampler','-1')   #This Sampler can be used to initialize the optimization initial points (e.g. partially replace the <initial> blocks for some variables)
+    AdaptiveSampler.__init__(self)
+    ## Instance Variable Initialization
+    # public
+    # _protected
+    self._seed = None           # random seed to apply
+    self._minMax = 'min'        # maximization or minimization?
+    self._activeTraj = []       # tracks live trajectories
+    self._cancelledTraj = {}    # tracks cancelled trajectories, and reasons
+    self._convergedTraj = {}    # tracks converged trajectories, and values obtained
+    self._numRepeatSamples = 1  # number of times to repeat sampling (e.g. denoising)
+    self._objectiveVar = None   # objective variable for optimization
+    self._initialValues = None  # initial variable values (trajectory starting locations), list of dicts
+    self._variableBounds = None # dictionary of upper/lower bounds for each variable (may be inf?)
+    self._trajCounter = 0       # tracks numbers to assign to trajectories
+    self._initSampler = None    # sampler to use for picking initial seeds
+    self._constraintFunctions = []
```

Comment:
docstring

Reply:
(even if it is clear that it is the list of the contraints functions)

Reply:
done; has this been added to the standards?